### PR TITLE
Add support for running unit tests against multiple versions of tsc

### PR DESCRIPTION
**Pull request by boyanio**
I have added a helper class `TscInstaller` to install, using npm, different versions of TypeScript and run the unit tests against each version of the compiler.

### CODE REVIEW

File: test/tsc-installer.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+const { execSync } = require('child_process');
+const { join } = require('path');
+const { readdirSync, lstatSync } = require('fs');
+
+const tscListDir = join(__dirname, 'tsc-list/');
+
+class TscInstaller {
```
Comment (gilamran) [inline test/tsc-installer.js]: There's no internal state to this class, so you might want to change it to functions only?

Comment (boyanio) [reply test/tsc-installer.js]: Good idea

**Comment by boyanio** [comment]
Good ideas, I will fix them. It seems there is an issue with the build due to permissions an spawn, so maybe I will have to change my approach a bit.

**Comment by boyanio** [comment]
It seems TravisCI has some issues with new folders. I tried creating a new folder in different ways (a static one, a dynamic one), but yet I get `spawn EACCESS` errors. Only modules created in the root `node_modules` seem to work fine

**Comment by gilamran** [comment]
I cloned your repo. and found that when you do npm install with `--no-bin-links` the compiler starts with `undefined` `stdout`

**Comment by boyanio** [comment]
Thanks! I removed this one, as initially I had a different approach. Now it seems to work fine, however some tests time out once in a while - it does not seem consistent

**Comment by gilamran** [comment]
You can try and increase the timeout?

**Comment by boyanio** [comment]
I did that, but then some tests fail for some reason. I don't think the issue is in the timeout. Plus, it works fine on my Windows machine, but it fails almost every time on the build server. I will try to look deeper inside the tests to find out the real reason

**Pull request closed without merging**